File: examples/v2/custom-vcs/go/main.go
package main

import (
	integrations "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/integrations"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		vcsSuffix := cfg.Get("vcsSuffix")
		if vcsSuffix == "" {
			vcsSuffix = "dev"
		}
		baseUrl := cfg.Get("baseUrl")
		if baseUrl == "" {
			baseUrl = "https://git.example.invalid"
		}
		envRef := cfg.Get("envRef")
		if envRef == "" {
			envRef = "organization/vcs-credentials"
		}

		integration, err := integrations.NewCustomVCSIntegration(ctx, "integration", &integrations.CustomVCSIntegrationArgs{
			OrgName:     pulumi.String(serviceOrg),
			Name:        pulumi.String("v2-custom-vcs-" + vcsSuffix),
			BaseUrl:     pulumi.String(baseUrl),
			VcsType:     pulumi.String("gitea"),
			Environment: pulumi.String(envRef),
		})
		if err != nil {
			return err
		}

		repository, err := integrations.NewCustomVCSRepository(ctx, "repository", &integrations.CustomVCSRepositoryArgs{
			OrgName:       pulumi.String(serviceOrg),
			IntegrationId: integration.IntegrationId,
			Name:          pulumi.String("example-repo-" + vcsSuffix),
			DisplayName:   pulumi.String("Example Repository"),
		})
		if err != nil {
			return err
		}

		ctx.Export("integrationId", integration.IntegrationId)
		ctx.Export("repositoryId", repository.ID())
		return nil
	})
}

File: examples/v2/custom-vcs/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
vcs_suffix = config.get("vcsSuffix") or "dev"
base_url = config.get("baseUrl") or "https://git.example.invalid"
env_ref = config.get("envRef") or "organization/vcs-credentials"

integration = ps_v2.integrations.CustomVCSIntegration(
    "integration",
    org_name=service_org,
    name=f"v2-custom-vcs-{vcs_suffix}",
    base_url=base_url,
    vcs_type="gitea",
    environment=env_ref,
)

repository = ps_v2.integrations.CustomVCSRepository(
    "repository",
    org_name=service_org,
    integration_id=integration.integration_id,
    name=f"example-repo-{vcs_suffix}",
    display_name="Example Repository",
)

pulumi.export("integrationId", integration.integration_id)
pulumi.export("repositoryId", repository.id)
